### PR TITLE
Improving handling of package list object names

### LINKED ISSUE #61: Improve provenance of srcref objects wrapped in package objects
Discovered when evaluating the [`tern`](https://github.com/insightsengineering/tern) package, aliases of values of lists provided by a package were tied to their names in the list, not their originating srcref it exists.

```r
# pkg
#' @export
fn <- function() "hello, world"

#' @export
obj <- list(x = fn)
```

```r
ttdf <- covtracer::test_trace_df(pkg_cov)
```

Would return srcrefs for an alias called "x", even though the code for `fn` stems from the bound object `fn`. Instead we should avoid masking srcrefs within lists.

### PR DESCRIPTION
**Pull request by dgkf**
Closes #61

This will preserve originating srcrefs for objects that get reintroduced into package list objects, which should address the wrapping of the function `format_count_fractions` as an element of `.a_summary_numeric_formats`.

### CODE REVIEW

File: R/srcref_df.R [hunk 1]
```diff
@@ -250,8 +250,8 @@ test_trace_mapping <- function(x) {
   if (!any(has_tests)) {
     empty_result <- matrix(
       integer(0L),
-      ncol = 4L,
-      dimnames = list(c(), c("test", "depth", "i", "trace"))
+      ncol = 5L,
+      dimnames = list(c(), c("test", "call", "depth", "i", "trace"))
     )
```
Comment (dgkf) [inline R/srcref_df.R:255]: This is unrelated to this PR, but is a necessary for use with the upcoming [PR to `covr`](https://github.com/dgkf/covr/tree/dgkf-min-dup-tests)

Comment (dgkf) [reply R/srcref_df.R]: Decided to address this most holistically, case-dependent on covr version.

**Comment by dgkf** [comment]
Still need to address CI failures. Currently it's not quite clear to me why package install is failing for this one testing package. Haven't been able to reproduce locally.

File: tests/testthat/setup_test_packages.R [hunk 2]
```diff
@@ -18,34 +18,33 @@ dir.create(lib <- tempfile("ct_"), recursive = TRUE)
...
 # install our testing packages into a temp directory
+install.packages(
+  as.character(pkg_dirs),
+  type = "source",
+  lib = lib,
+  repos = NULL,
+  dependencies = FALSE,
+  INSTALL_opts = c("--with-keep.source", "--install-tests"),
+  quiet = TRUE
+)
```
Comment (dgkf) [inline tests/testthat/setup_test_packages.R:36]: I had the bright idea of cleaning up our testing setup, but it sent me down a rabbit hole of rediscovering how a package with no source code crashes our test suite.

File: R/srcrefs.R [hunk 3]
```diff
@@ -85,7 +85,11 @@ srcrefs.list <- function(x, ..., srcref_names = NULL, breadcrumbs = character())
   # it could lead to unexpected results returned by the mapper. Thus, if
   # list was the class used for dispatch, we remove other classes.
   x <- unclass(x)
-  flat_map_srcrefs(x, ns = srcref_names, breadcrumbs = breadcrumbs)
+
+  # propagate srcref names as element names
+  if (!is.null(srcref_names)) names(x) <- rep_len(srcref_names, length(x))
```
[the remark this answers is not in the document]
Comment (maksymiuks) [reply R/srcrefs.R:90]: I see. What gonna happen if the list won't have names though? Do we provide some custom ones?

**Review by maksymiuks — APPROVED** [review]
Looks really good! Approved provided we have the handling of lists without names covered.

**Pull request merged**